modules/apps/frontend-js/frontend-js-module-launcher/src/main/java/com/liferay/frontend/js/module/launcher/internal/JSModuleLauncherImpl.java

```diff
@@ -0,0 +1,142 @@
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.frontend.js.module.launcher.internal;
+
+import com.liferay.frontend.js.module.launcher.JSModuleLauncher;
+import com.liferay.petra.string.StringBundler;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.servlet.taglib.aui.ScriptData;
+import com.liferay.portal.kernel.util.WebKeys;
+
+import java.io.IOException;
+import java.io.Writer;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * @author Iván Zaera Avellón
+ */
+@Component(service = JSModuleLauncher.class)
+public class JSModuleLauncherImpl implements JSModuleLauncher {
+
+	@Override
+	public void appendPortletScript(
+		HttpServletRequest httpServletRequest, String portletId,
+		String javascriptModule, String javascriptVariable,
+		String javascriptCode) {
+
+		_appendScriptData(
+			httpServletRequest,
+			_getScriptBody(
+				javascriptModule, javascriptVariable, javascriptCode),
+			portletId);
+	}
+
+	@Override
+	public void appendScript(
+		HttpServletRequest httpServletRequest, String javascriptModule,
+		String javascriptVariable, String javascriptCode) {
+
+		_appendScriptData(
+			httpServletRequest,
+			_getScriptBody(
+				javascriptModule, javascriptVariable, javascriptCode),
+			null);
+	}
+
+	@Override
+	public void writePortletScript(
+		Writer writer, String portletId, String javascriptModule,
+		String javascriptVariable, String javascriptCode) {
+
+		_writeScriptData(
+			writer,
+			_getScriptBody(
+				javascriptModule, javascriptVariable, javascriptCode),
+			portletId);
+	}
+
+	@Override
+	public void writeScript(
+		Writer writer, String javascriptModule, String javascriptVariable,
+		String javascriptCode) {
+
+		_writeScriptData(
+			writer,
+			_getScriptBody(
+				javascriptModule, javascriptVariable, javascriptCode),
+			null);
+	}
+
+	private void _appendScriptData(
+		HttpServletRequest httpServletRequest, String content,
+		String portletId) {
+
+		ScriptData scriptData = (ScriptData)httpServletRequest.getAttribute(
+			WebKeys.AUI_SCRIPT_DATA);
+
+		if (scriptData == null) {
+			scriptData = new ScriptData();
+
+			httpServletRequest.setAttribute(
+				WebKeys.AUI_SCRIPT_DATA, scriptData);
+		}
+
+		scriptData.append(
+			portletId, content, StringPool.BLANK, ScriptData.ModulesType.ES6);
+	}
+
+	private String _getScriptBody(
+		String javascriptModule, String javascriptVariable,
+		String javascriptCode) {
+
+		StringBundler javascriptSB = new StringBundler(10);
+
+		javascriptSB.append("(function() {");
+
+		javascriptSB.append("window[");
+		javascriptSB.append("Symbol.for('__LIFERAY_WEBPACK_GET_MODULE__')]('");
+		javascriptSB.append(javascriptModule);
+		javascriptSB.append("').then((");
+		javascriptSB.append(javascriptVariable);
+		javascriptSB.append(") => {");
+		javascriptSB.append(javascriptCode);
```

Comment:
Feels like a lot of times we'll have `javascriptCode === javascriptVariable.default([config])`. 

We might want to provide something for this common case. Should also be trivial to allow named exports (non-default).  Finally, multi-parameter might look weirder as in `Something.default(paramA, paramB)` but I think that should be doable as well. 

In essence, feels like providing some launching semantics can alleviate the pain of constructing that `javascriptCode` and keeping things in sync throughout the process. 

We can get this through and build on top, since I guess the generic API won't change and is a good escape hatch for things we didn't foresee.

Reply:
I wanted to do that automatically (probably in the webpack_federation.js module) but I still don't have it clear because the handling of the ES6 interoperation has a lot of subtleties, as we already saw in the bundler. 

So I prefer to wait a bit and see more cases...

Reply:
Mmm, in fact it can be done in user's code, like you would do in `node`. For example, instead of:

```java
_jsModuleLauncher.writeScript(
	httpServletResponse.getWriter(), 
	"frontend-js-alert-support-web",
	"AlertProvider", 
	"AlertProvider.default()"
);
```

we can do:

```java
_jsModuleLauncher.writeScript(
	httpServletResponse.getWriter(), 
	"frontend-js-alert-support-web",
	"{default: AlertProvider}", 
	"AlertProvider()"
);
```
which is the same syntax you must use in `node` when you want to use ES modules from pure CommonJS modules (unless the provider has exported the same under  `module.exports` and `module.exports.default`, in which case, the other syntax would work too.

Reply:
I have added a final commit to make the API simpler to use.